integration-pod-api-client/src/main/java/org/symphonyoss/integration/pod/api/client/IntegrationAuthApiClient.java

```diff
@@ -200,10 +209,99 @@ public UserAuthorizationData getUserAuthData(String sessionToken, String integra
    */
   private void checkParam(Object param, String paramName) throws RemoteApiException {
     if (param == null) {
-      String reason = logMessage.getMessage(MISSING_PARAMETER_WHEN_CALLING, paramName, GET_USER_AUTH_DATA);
+      String reason =
+          logMessage.getMessage(MISSING_PARAMETER_WHEN_CALLING, paramName, GET_USER_AUTH_DATA);
       String solution = logMessage.getMessage(MISSING_PARAMETER_WHEN_CALLING_SOLUTION, paramName);
       throw new RemoteApiException(HTTP_BAD_REQUEST_ERROR, reason, solution);
     }
   }
 
+  /**
+   * Save application authentication data.
+   * @param sessionToken Session authentication token.
+   * @param integrationId Integration identifier
+   * @param appToken Pair of tokens - application one and Symphony one.
+   * @throws UnauthorizedUserException User credentials not provided
+   * @throws ForbiddenAuthException User not authorized to retrieve user authentication
+   */
+  public void saveAppAuthenticationToken(String sessionToken, String integrationId,
+      AppToken appToken) {
+    try {
+      checkAuthToken(sessionToken);
+
+      checkParam(integrationId, INTEGRATION_ID);
```

Comment:
The 'checkParam' method might raise an AuthenticationException rather than RemoteApiException.

Using the current approach we would disappear with the parameter validation message.